### DISCUSSION TITLE
Way to declare SET

### DISCUSSION THREAD

**Discussion by naveedpash** [librasn/rasn#38] [Q&A] [answered]
Hey,
(maybe a silly question)

If I have the following ASN.1

```asn.1
Mode-selector ::= SET { mode-value [0] IMPLICIT INTEGER { x410-1984-mode (0), normal-mode (1) } }
```

how do I code this in rust? is below an adequate example?

```rust
#[derive(Debug, AsnType, Encode, Decode)]
#[rasn(tag(universal, 17))]
pub struct ModeSelector {
  #[rasn(tag(0))]
  mode_value: u8
}
```

**Comment by XAMPPRocky** [comment, marked as answer]
Thank you for your question! I have not written the documentation for `SET` types yet, as I haven't released it yet. But how you do it is very simple, you just add `#[rasn(set)]` to the container like `#[rasn(choice)]`.

```rust
#[derive(Debug, AsnType, Encode, Decode)]
#[rasn(tag(universal, 17))]
#[rasn(set)]
pub struct ModeSelector {
  #[rasn(tag(0))]
  mode_value: u8
}
```

**Reply by naveedpash** [reply]
Awaiting your next release, thank you for this amazing crate!
